src/lib.rs

```diff
@@ -7,36 +7,23 @@ extern crate rand;
 
 use rand::{ChaChaRng, Rng, SeedableRng};
```

Comment:
Fun fact, this was such a big change that Github hid it by default.

Reply:
Yeah, I noticed. 😳 

I'm not surprised. The existing implementation was pretty toy, and then I had to come up with a completely new way to represent the actual main logic.